File: Producto/routes-service/src/main/java/com/tagok/routes_service/domain/tarifa/Cruce.java
package com.tagok.routes_service.domain.tarifa;

import java.math.BigDecimal;
import java.time.LocalDateTime;

public sealed interface Cruce permits CrucePortico, CruceTramo 
{
    String codigo();
    String nombre();
    String autopista();
    TipoTarifa tipoTarifa();
    BigDecimal valor();
    LocalDateTime horaFechaCruce();
}

File: Producto/tag-ok-app/app/src/main/java/com/tagok/app/data/dto/Cruce.kt
package com.tagok.app.data.dto

import kotlinx.serialization.Serializable

@Serializable
data class Cruce(
    val porticoId: Long,
    val codigo: String,
    val nombre: String? = null,
    val autopista: String? = null,
    val tarifa: String? = null,
    val valor: Double = 0.0)
    // horaFechaCruce omitido — solo necesitamos los valores monetarios
